File: backend/types/events.py
"""Pydantic models for viral events, deep dives, and audits."""

from pydantic import BaseModel, Field

from backend.types.enums import ConfidenceLevel, EventCategory


class ViralEvent(BaseModel):
    """A single viral AI event detected by a research engine."""

    headline: str
    category: EventCategory
    impact_rating: int = Field(ge=1, le=10)
    confidence: ConfidenceLevel
    source: str


class DeepDive(BaseModel):
    """A strategic deep-dive analysis section."""

    title: str
    priority: str
    summary: str
    key_findings: list[str]


class CompletenessAudit(BaseModel):
    """Audit of research completeness and signal coverage."""

    verified_signals: int
    sources_checked: int
    confidence_score: float = Field(ge=0.0, le=1.0)
    gaps: list[str]

File: rust-backend/src/types/events.rs
use serde::{Deserialize, Serialize};

use super::enums::{ConfidenceLevel, EventCategory};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ViralEvent {
    pub headline: String,
    pub category: EventCategory,
    pub impact_rating: i32,
    pub confidence: ConfidenceLevel,
    pub source: String,
    pub summary: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DeepDive {
    pub title: String,
    pub priority: String,
    pub summary: String,
    pub key_findings: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CompletenessAudit {
    pub verified_signals: i32,
    pub sources_checked: i32,
    pub confidence_score: f64,
    pub gaps: Vec<String>,
}
